Objects/memoryobject.c

```diff
@@ -3303,10 +3305,51 @@ typedef struct {
     const char *it_fmt;
 } memoryiterobject;
 
+#define _PyMemoryViewIter_CAST(PTR) ((memoryiterobject *)(PTR))
+
+static PyObject *
+memoryiter_new(PyObject *self, int reversed)
+{
+    if (!PyMemoryView_Check(self)) {
+        PyErr_BadInternalCall();
+        return NULL;
+    }
+
+    PyMemoryViewObject *sequence = _PyMemoryView_CAST(self);
+    const Py_buffer *view = &sequence->view;
+    if (view->ndim == 0) {
+        PyErr_SetString(PyExc_TypeError, "invalid indexing of 0-dim memory");
+        return NULL;
+    }
+    else if (view->ndim != 1) {
+        PyErr_SetString(PyExc_NotImplementedError,
+                        "multi-dimensional sub-views are not implemented");
+        return NULL;
+    }
+    const char *fmt = adjust_fmt(view);
+    if (fmt == NULL) {
+        return NULL;
+    }
+
```

Comment:
A sanity check would be good here
```suggestion
assert(reversed == 0 || reversed == 1);
```

Reply:
Honestly, I'm not a fan of that though I'm not sure if this is common in the rest of the codebase. Alternatively, I think changing the type to `bool` would convene the intent.

Reply:
Never mind, the rest of the code base seems to also do it in general. So I'll add that assertion as well.